app/src/main/java/de/westnordost/streetcomplete/quests/bus_stop_name/AddBusStopName.java

```diff
@@ -0,0 +1,50 @@
+package de.westnordost.streetcomplete.quests.bus_stop_name;
+
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+
+import java.util.Map;
+
+import javax.inject.Inject;
+
+import de.westnordost.streetcomplete.R;
+import de.westnordost.streetcomplete.data.osm.SimpleOverpassQuestType;
+import de.westnordost.streetcomplete.data.osm.changes.StringMapChangesBuilder;
+import de.westnordost.streetcomplete.data.osm.download.OverpassMapDataDao;
+import de.westnordost.streetcomplete.quests.AbstractQuestAnswerFragment;
+
+public class AddBusStopName extends SimpleOverpassQuestType
+{
+
+	@Inject public AddBusStopName(OverpassMapDataDao overpassServer) { super(overpassServer); }
+
+	@Override protected String getTagFilters()
+	{
+		return " nodes, ways with !name and noname != yes" +
+			   " and (public_transport=platform or (highway=bus_stop and public_transport!=stop_position))";
+	}
+
+	@Override public AbstractQuestAnswerFragment createForm()
+	{
+		return new AddBusStopNameForm();
+	}
+
+	@Override public void applyAnswerTo(Bundle answer, StringMapChangesBuilder changes)
+	{
+		if(answer.getBoolean(AddBusStopNameForm.NO_NAME))
+		{
+			changes.add("noname", "yes");
+			return;
+		}
+
+		String name = answer.getString(AddBusStopNameForm.NAME);
+		if(name != null) changes.add("name", name);
+	}
+
+	@Override public String getCommitMessage() { return "Determine bus stop names"; }
+	@Override public int getIcon() { return R.drawable.ic_quest_bus_stop_name; }
+	@Override public int getTitle(@NonNull Map<String, String> tags)
+	{
+		return R.string.quest_busStopName_title;
```

Comment:
Right now this includes any kind of public transport platform. It would be odd if the quest asked that question for a tram, railway or metro stop. The question should be different for every different type of included public transport platform. I suggest to only include bus and tram stops and then have two strings. See 
https://github.com/westnordost/StreetComplete/blob/master/app/src/main/java/de/westnordost/streetcomplete/quests/bus_stop_shelter/AddBusStopShelter.java#L24-L26